src/System.Linq.Parallel/src/System/Linq/Parallel/Channels/AsynchronousChannel.cs

```diff
@@ -395,7 +401,7 @@ private void FlushCachedChunk()
         //     True if an item was found, false otherwise.
         //
 
-        internal bool TryDequeue(ref T item)
+        internal bool TryDequeue([AllowNull] ref T item)
```

Comment:
💭 Should this also have `MaybeNullWhen(false)`?

Reply:
I think because it is `ref` not `out` parameter it would keep the value whatever it had with false return

Reply:
Added MaybeNullWhen(false), thanks